src/Disks/ObjectStorages/S3/registerDiskS3.cpp

```diff
@@ -65,7 +65,35 @@ void checkReadAccess(const String & disk_name, IDisk & disk)
         throw Exception("No read access to S3 bucket in disk " + disk_name, ErrorCodes::PATH_ACCESS_DENIED);
 }
 
-void checkRemoveAccess(IDisk & disk) { disk.removeFile("test_acl"); }
+void checkRemoveAccess(IDisk & disk)
+{
+    disk.removeFile("test_acl");
+}
+
+bool checkBatchRemoveIsMissing(S3ObjectStorage & storage, const String & key)
+{
+    String path = key + "/test_acl";
+    auto file = storage.writeObject(path, WriteMode::Rewrite);
+    try
+    {
+        file->write("test", 4);
+    }
+    catch (...)
+    {
+        file->finalize();
+        return false; /// We don't have write access, therefore no information about batch remove.
```

Comment:
The other two returns (l. 90, l. 94) are okay. Here, even writing the file fails (for whatever reason) - we cannot know if the storage supports batch delete or not. A more conservative approach would be to return true and to set support_batch_delete = false (l. 132). Alternatively, we could also let the exception simply bubble up as the error is unexpected.

Reply:
Yes, it will bubble up here: https://github.com/ClickHouse/ClickHouse/pull/37882/files#diff-592e84ce237a5530062359e33672a682d53c2ff5434ba9a7e69d958b3ba90448R152

Reply:
Okay, I am convinced 😄